src/Adapter/Shop/MaintenanceConfiguration.php

```diff
@@ -61,24 +51,40 @@ public function getConfiguration()
      */
     public function updateConfiguration(array $configuration)
     {
-        if ($this->validateConfiguration($configuration)) {
-            $this->configuration->set('PS_SHOP_ENABLE', $configuration['enable_shop']);
-            $this->configuration->set('PS_MAINTENANCE_IP', $configuration['maintenance_ip']);
-            $this->configuration->set('PS_MAINTENANCE_TEXT', $configuration['maintenance_text'], null, ['html' => true]);
+        $shopConstraint = null;
+        if (!$this->shopContext->isAllShopContext()) {
```

Comment:
This part will probably be movable into an abstract class when you start integrating this behaviour in other forms (not needed for now)

Reply:
Oh bu I didn't notice that you already introduced an abstract `MultistoreConfigurator` The maybe it could integrate this part and return the `ShopContraint` object along with update input array (or in two different methods)